walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(events.isEmpty() && request.getDuration() <= TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    
+    List<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    List<Event> eventList = new ArrayList<>(events);
+    Event firstEvent = eventList.get(0);
+    Event lastEvent = eventList.get(eventList.size() - 1);
+
+    int eventsSkipped = 0;
+
+    if(eventList.size() == 1) {
+      if(!requestHasAttendees(request, firstEvent)) {
+        eventsSkipped++;
+      }
+    }
+
+    // Add the first time slot, from the start of the day to the start of the first event,
+    // only if there's no event that starts the day.
+    if(dayStartsWithEvent(eventList) && requestHasAttendees(request, firstEvent)) {
```

Comment:
You can assign `requestHasAttendees(request, firstEvent)` to a variable  before the if above to prevent calling it twice.